pyramid_hypernova/request.py

```diff
@@ -42,23 +43,23 @@ def __init__(self, job_group, url, json_encoder, synchronous, request_headers):
     def send(self):
         """ Query Hypernova """
         job_str = self.json_encoder.encode(create_jobs_payload(self.job_group))
-        job_bytes = job_str.encode('utf-8')
+        self.job_bytes = job_str.encode('utf-8')
 
-        request_headers = dict(self.request_headers)
-        request_headers['Content-Type'] = 'application/json'
+        self.request_headers = dict(self.request_headers)
+        self.request_headers['Content-Type'] = 'application/json'
 
         if self.synchronous:
-            self.response = requests.post(
-                url=self.url,
-                headers=request_headers,
-                data=job_bytes,
-            )
+            # do nothing! requests.post() will throw an HTTPError if there's no healthy SSR
+            # upstream. we're not expecting this method to ever throw an exception,
+            # so make synchronous SSR requests in json() instead, where we're equipped to
```

Comment:
just confirming, that `where we're equipped to catch and deal with them` refers to consumers of this library (ie. services) right?

Reply:
nope, `batch` catches the exception and returns the fallback response transparently to the service https://github.com/Yelp/pyramid-hypernova/blob/master/pyramid_hypernova/batch.py#L119